### PR TITLE
LinkedListQueue Added

### LINKED ISSUE #49: Add Linked Lists
#### Description of the problem
Currently the package does not have Queues or Linked Lists. 
While working with issue #22-Add Queue  
It was discovered that to add a Queue which is efficient enough to perform popleft() action in O(1) time we need linked lists to implement Queues just like the collections.deque implementaion of a Queue.

Since the original goal of issue #22 was to reduce dependencies on other libraries, LInked List must be added to pydatastructs to ensure an efficient implementation of Queues.

#### Example of the problem

>>> sample_queue = Queue()
>>> for i in range(N) # N be a large number
...        sample_queue.append('#')
...
>>> sample_queue.popleft() # Takes O(N) time
#

#### References/Other comments

See the original issue -
https://github.com/codezonediitj/pydatastructs/issues/22

Read about the efficiency issue here -
https://www.geeksforgeeks.org/deque-in-python/
https://stackoverflow.com/questions/55578874/dequeue-complexity-on-using-python-list-as-a-queue

### PR DESCRIPTION
**Pull request by prshnt19**
#### References to other Issues or PRs or Relevant literature

FIxes #49 #113

### CODE REVIEW

File: pydatastructs/miscellaneous_data_structures/queue.py [hunk 1]
```diff
@@ -51,6 +51,11 @@ def __new__(cls, implementation='array', **kwargs):
             return ArrayQueue(
                 kwargs.get('items', None),
                 kwargs.get('dtype', int))
+        elif implementation == 'linkedlist':
+            return LinkedListQueue(
+                # kwargs.get('items', None),
+                # kwargs.get('dtype', LinkedListNode)
```
Comment (czgdp1807) [inline pydatastructs/miscellaneous_data_structures/queue.py]: Why is this commented out? Can we keep the API similar to the `ArrayQueue`?

File: pydatastructs/miscellaneous_data_structures/queue.py [hunk 2]
```diff
@@ -64,7 +69,9 @@ def popleft(self, *args, **kwargs):
 
     @property
     def is_empty(self):
-        return None
+        raise NotImplementedError(
+            "This is an abstract method.")
+
```
Comment (czgdp1807) [inline pydatastructs/miscellaneous_data_structures/queue.py]: `return None` was used to show the abstractness. I think the change is fine to let the user know they are trying to access something that isn't defined.

Comment (prshnt19) [reply pydatastructs/miscellaneous_data_structures/queue.py]: Also, it improves Codecov.

File: pydatastructs/miscellaneous_data_structures/queue.py (hunk 1 above)
Comment (prshnt19) [reply pydatastructs/miscellaneous_data_structures/queue.py]: Made `LinkedListQueue` analogous to `ArrayQueue`. Now they have similar API.

**Comment by czgdp1807** [comment]
@manaswinidas Please see if the tests are conceptually correct.

**Pull request merged**